src/main/java/org/embulk/input/zendesk/clients/ZendeskRestClient.java

```diff
@@ -0,0 +1,188 @@
+import org.eclipse.jetty.client.api.Request;
+import org.eclipse.jetty.client.api.Response;
+import org.eclipse.jetty.http.HttpMethod;
+import org.embulk.config.ConfigException;
+import org.embulk.input.zendesk.ZendeskInputPluginDelegate;
+import org.embulk.input.zendesk.utils.ZendeskConstants;
+import org.embulk.input.zendesk.utils.ZendeskUtils;
+import org.embulk.spi.Exec;
+import org.embulk.util.retryhelper.jetty92.Jetty92ResponseReader;
+import org.embulk.util.retryhelper.jetty92.Jetty92RetryHelper;
+import org.embulk.util.retryhelper.jetty92.Jetty92SingleRequester;
+import org.slf4j.Logger;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+
+public class ZendeskRestClient implements AutoCloseable
+{
+    private static final Logger logger = Exec.getLogger(ZendeskRestClient.class);
+    private static RateLimiter rateLimiter;
+
+    private final Jetty92RetryHelper retryHelper;
+    private final ZendeskInputPluginDelegate.PluginTask task;
+
+    public ZendeskRestClient(final Jetty92RetryHelper retryHelper, final ZendeskInputPluginDelegate.PluginTask task)
+    {
+        this.retryHelper = retryHelper;
+        this.task = task;
+    }
+
+    public <T> T doGet(final String url, final Jetty92ResponseReader<T> responseReader)
+    {
+        T result = retryHelper.requestWithRetry(responseReader, new ZendeskV2Jetty92SingleRequester(url, buildAuthHeader()));
+        getRateLimiter(responseReader).acquire();
+        return result;
+    }
+
+    @Override
+    public void close()
+    {
+        retryHelper.close();
+    }
+
+    private static class ZendeskV2Jetty92SingleRequester extends Jetty92SingleRequester
+    {
+        private final String url;
+        private final ImmutableMap<String, String> headers;
+
+        ZendeskV2Jetty92SingleRequester(final String url, final ImmutableMap<String, String> headers)
+        {
+            this.url = url;
+            this.headers = headers;
+        }
+
+        @Override
+        public void requestOnce(final HttpClient client, final Response.Listener responseListener)
+        {
+            final Request request = client.newRequest(url).method(HttpMethod.GET);
+
+            if (headers != null) {
+                for (final Map.Entry<String, String> entry : headers.entrySet()) {
+                    request.header(entry.getKey(), entry.getValue());
+                }
+            }
+
+            logger.info(">>> {} {}{}", request.getMethod(), request.getURI().getPath(),
+                    request.getURI().getQuery() != null ? "?" + request.getURI().getQuery() : "");
+            request.send(responseListener);
+        }
+
+        /**
+         * Analyze response whether the failure response is retryable or not
+         *
+         * Only retry if reaches the API limitation
+         */
+        @Override
+        protected boolean isResponseStatusToRetry(final Response response)
```

Comment:
This method is not on par with Ruby version, please double check.

Reply:
For example, old code will retry `404`. `422` will only be retried when `$.description` starts with `Too recent start_time.`